airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/generate_erd/pipeline.py

```diff
@@ -0,0 +1,266 @@
+        """
+        response = self._model.generate_content(f"{system} {source_desc} {task}")
+        md = MarkdownIt("commonmark")
+        tokens = md.parse(response.text)
+        response_json = json.loads(tokens[0].content)
+        return response_json
+
+
+class GenerateDbmlSchema(Step):
+    context: ConnectorContext
+
+    title = "Generate DBML file from discovered catalog and erd_relation"
+
+    def __init__(self, context: PipelineContext) -> None:
+        super().__init__(context)
+
+    def _get_catalog(self, catalog_path: Path) -> AirbyteCatalog:
+        with open(catalog_path, "r") as file:
+            try:
+                return AirbyteCatalog.parse_obj(json.loads(file.read()))
+            except json.JSONDecodeError as error:
+                raise ValueError(f"Could not read json file {catalog_path}: {error}. Please ensure that it is a valid JSON.")
+
+    def _get_relationships(self, path: Path) -> Relationships:
+        if not path.exists():
+            return {"streams": []}
+
+        with open(path, "r") as file:
+            return json.load(file)  # type: ignore  # we assume the content of the file matches Relationships
+
+    async def _run(self, connector_to_discover: Container = None) -> StepResult:
+        connector = self.context.connector
+        python_path = connector.code_directory
+
+        catalog = self._get_catalog(_get_discovered_catalog_file(connector.code_directory))
+        source = Source(python_path)
+        confirmed_relationships = self._get_relationships(_get_confirmed_relationships_file(connector.code_directory))
+        estimated_relationships = self._get_relationships(_get_estimated_relationships_file(connector.code_directory))
+        database = DbmlAssembler().assemble(
+            source,
+            catalog,
+            RelationshipsMerger().merge(estimated_relationships, confirmed_relationships)
+        )
+
+        with open(_get_dbml_file(connector.code_directory), "w") as f:
+            f.write(DefaultDBMLRenderer.render_db(database))
+
+        return StepResult(step=self, status=StepStatus.SUCCESS)
+
+
+class UploadDbmlSchema(Step):
+    context: ConnectorContext
+
+    title = "Upload DBML file to dbdocs.io"
+
+    def __init__(self, context: PipelineContext) -> None:
+        super().__init__(context)
+
+    async def _run(self, connector_to_discover: Container = None) -> StepResult:
+        connector = self.context.connector
+        source_dbml_content = open(_get_dbml_file(connector.code_directory)).read()
+
+        dbdocs_container = await (
+            self.dagger_client.container()
+            .from_("node:lts-bullseye-slim")
+            .with_exec(["npm", "install", "-g", "dbdocs"])
+            .with_env_variable("DBDOCS_TOKEN", DBDOCS_TOKEN)
+            .with_workdir("/airbyte_dbdocs")
+            .with_new_file("/airbyte_dbdocs/dbdocs.dbml", contents=source_dbml_content)
```

Comment:
I've seen some weird cases where the dbdocs diagram wasn't updated as part of the airbyte-ci command but running the same command locally would update the dbdocs project. Can we trust that if `source_dbml_content` is different, the cache won't be triggered or is it safer to add `.with_env_variable("CACHEBUSTER", str(uuid.uuid4()))`?